Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4244,7 +4244,7 @@ void WebPageProxy::commitProvisionalPage(FrameIdentifier frameID, FrameInfoData&
 
     RefPtr<WebFrameProxy> mainFrameInPreviousProcess = m_mainFrame;
 
-    ASSERT(m_process.ptr() != &m_provisionalPage->process());
+    ASSERT(m_process->processPool().configuration().usesSingleWebProcess() || m_process.ptr() != &m_provisionalPage->process());
```

Comment:
I don't think we should have ever ended up this code path in single process mode. This code path occurs after process-swapping on navigation. When you're using a single WebProcess, you cannot process swap.

Odds are that the proper fix should happen earlier, to make sure we don't end up creating a provisional page in the first place.

Reply:
@cdumez Thanks for the review. Also, I am sorry for the delay in responding.

> Odds are that the proper fix should happen earlier, to make sure we don't end up creating a provisional page in the first place.

To avoid creating a provisional page, it may be necessary to modify the NetworkProcess side.
The value of the browserContextGroupSwitchDecision determined at [NetworkResourceLoader::sendDidReceiveResponsePotentiallyInNewBrowsingContextGroup()](https://github.com/WebKit/WebKit/blob/main/Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp#L993) whether process swapping was executed or not.

I will try to create a modification in that the flowsingContextGroupSwitchDecision is `StayInGroup` in the case of SingleWebProcess to avoid creating a provisional page.